packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -262,6 +262,16 @@ export const BlocksDropdown = () => {
     }
   };
 
+  React.useEffect(() => {
+    const selectedBlocks = Object.keys(blocks).filter((key) => {
+      return isBlockActive(editor, blocks[key].matchNode);
+    });
+
+    if (selectedBlocks.length === 1) {
+      setBlockSelected(selectedBlocks[0]);
+    }
```

Comment:
Would find by more appropriate because you want to know what the selectedBlock is and then you can say "hey stop looking when you've found it"?